### PR TITLE
Add the "Auto" theme option in setting

### PR DESCRIPTION
**Pull request by yell0wsuit**
https://github.com/Ryujinx/Ryujinx/assets/5692900/49125c81-6333-4d42-84fa-063606c1e820

(Need to fix the font color in the dropdown when the theme changes - leave that for another issue)

### CODE REVIEW

File: src/Ryujinx/App.axaml.cs [hunk 1]
```diff
@@ -111,5 +115,29 @@ private void ApplyConfiguredTheme()
...
+
+                default:
+                    return ThemeVariant.Default;
+            }
+        }
+
+        private ThemeVariant DetectSystemTheme()
+        {
+            var platformSettings = this.PlatformSettings;
+            var colorValues = platformSettings.GetColorValues();
```
Comment (gdkchan) [inline src/Ryujinx/App.axaml.cs:138]:
```suggestion
            var colorValues = PlatformSettings.GetColorValues();
```
And the `platformSettings` variable can be removed then. Or is there a special reason for having this variable?

File: src/Ryujinx/App.axaml.cs [hunk 2]
```diff
@@ -111,5 +115,29 @@ private void ApplyConfiguredTheme()
                 ShowRestartDialog();
             }
         }
+
+        // Convert PlatformThemeVariant to the expected ThemeVariant type
+        private ThemeVariant ConvertThemeVariant(PlatformThemeVariant platformThemeVariant)
+        {
+            switch (platformThemeVariant)
```
Comment (gdkchan) [inline src/Ryujinx/App.axaml.cs:122]: Can you turn that into a switch expression?

File: src/Ryujinx/App.axaml.cs [hunk 3]
```diff
@@ -111,5 +115,29 @@ private void ApplyConfiguredTheme()
                 ShowRestartDialog();
             }
         }
+
+        // Convert PlatformThemeVariant to the expected ThemeVariant type
```
Comment (gdkchan) [inline src/Ryujinx/App.axaml.cs:119]: If you want to include a comment here, it should be a proper XML doc comment, that will allow it to show on IDE.

File: src/Ryujinx/UI/Windows/MainWindow.axaml.cs [hunk 4]
```diff
@@ -87,6 +88,25 @@ public MainWindow()
             }
         }
 
+        // Event handler for detecting OS theme change when using "Follow OS theme" option
```
Comment (gdkchan) [inline src/Ryujinx/UI/Windows/MainWindow.axaml.cs:91]: Same comment about the XML docs.

File: src/Ryujinx/UI/Windows/MainWindow.axaml.cs [hunk 5]
```diff
@@ -87,6 +88,25 @@ public MainWindow()
...
+        private void OnPlatformColorValuesChanged(object? sender, PlatformColorValues e)
```
Comment (gdkchan) [inline src/Ryujinx/UI/Windows/MainWindow.axaml.cs:92]:
```suggestion
        private void OnPlatformColorValuesChanged(object sender, PlatformColorValues e)
```

File: src/Ryujinx/UI/Windows/MainWindow.axaml.cs [hunk 6]
```diff
@@ -87,6 +88,25 @@ public MainWindow()
...
+            if (Application.Current is App app)
+            {
+                app.ApplyConfiguredTheme();
+            }
+        }
+
+        protected override void OnClosed(EventArgs e)
+        {
+            base.OnClosed(e);
+            if (this.PlatformSettings != null)
```
Comment (gdkchan) [inline src/Ryujinx/UI/Windows/MainWindow.axaml.cs:103]: No need for the `this.` here and below, unless I'm missing something.

File: src/Ryujinx/App.axaml.cs (hunk 1 above)
Comment (yell0wsuit) [reply src/Ryujinx/App.axaml.cs:138]: Yeah, my bad. Just `PlatformSettings` is enough.

**Comment by IsaacMarovitz** [comment]
The CheckBox issue was reported upstream, and they were unable to reproduce it. https://github.com/amwx/FluentAvalonia/issues/366

File: src/Ryujinx/Assets/Locales/en_US.json [hunk 7]
```diff
@@ -298,6 +298,7 @@
   "GameListContextMenuToggleFavorite": "Toggle Favorite",
   "GameListContextMenuToggleFavoriteToolTip": "Toggle Favorite status of Game",
   "SettingsTabGeneralTheme": "Theme:",
+  "SettingsTabGeneralThemeAuto": "Follow OS theme",
```
Comment (IsaacMarovitz) [inline src/Ryujinx/Assets/Locales/en_US.json:301]: I think just saying "Auto" would be better here

**Comment by yell0wsuit** [comment]
Refactored the about dialog code. The theme icons seemingly use dark variant even when the OS theme is light.

Before:

Screenshot 2024-04-07 105342

After:

Screenshot 2024-04-07 105428

**Review by IsaacMarovitz — APPROVED** [review]
Lgtm

**Review by AcK77 — APPROVED** [review]

**Pull request merged**